core/src/broadcast_stage.rs

```diff
@@ -342,33 +343,33 @@ impl BroadcastStage {
         retransmit_slots_receiver: &RetransmitSlotsReceiver,
         socket_sender: &Sender<(Arc<Vec<Shred>>, Option<BroadcastShredBatchInfo>)>,
     ) -> Result<()> {
-        let timer = Duration::from_millis(100);
+        let mut sel = Select::new();
+        sel.recv(retransmit_slots_receiver);
+        let _ = sel.ready_timeout(Duration::from_millis(100))?;
 
         // Check for a retransmit signal
-        let mut retransmit_slots = retransmit_slots_receiver.recv_timeout(timer)?;
-        while let Ok(new_retransmit_slots) = retransmit_slots_receiver.try_recv() {
-            retransmit_slots.extend(new_retransmit_slots);
```

Comment:
My reading of the original code was that it was trying to dedup multiple copies of the same bank being submitted in succession. Is that unnecessary?

Reply:
good catch, moved to a `HashSet` on the receiving end